### PR TITLE
feat: add bl_chunk_size to LSTBIN_OPTS

### PR DESCRIPTION
**Pull request by steven-murray**
Adds `bl_chunk_size` to LSTBIN_OPTS

### CODE REVIEW

File: pipelines/h6c/idr2/v3/lstbin/lstbin-template.toml [hunk 1]
```diff
@@ -26,6 +26,11 @@ timeout = "48h"
 [LSTBIN_OPTS]                                                                                                                                                                                                                                                                                                                                               
 # ---- Parameters used by hera_opm to build the makeflow ----
 outdir = "${LSTAVG_OPTS:outdir}"
+{% if REDAVG -%}
+bl_chunk_size = 5000
+{% else %}
+bl_chunk_size = 0
+{% endif %}
```
Comment (jsdillon) [inline pipelines/h6c/idr2/v3/lstbin/lstbin-template.toml:34]: I wasn't aware you could do conditionals in tomls... can you explain in a comment how this works? Also, where is `REDAVG` defined, I don't immediately see it.

File: pipelines/h6c/idr2/v3/lstbin/lstbin-template.toml [hunk 2]
```diff
@@ -54,7 +59,7 @@ fileglob = "{night}/zen.{night}.?????.sum{{ DATA_EXTENSION }}.uvh5"
 [LSTAVG_OPTS]
 # Options for the notebook running -- i.e. the metrics and averaging.
 outdir = "/lustre/aoc/projects/hera/h6c-analysis/IDR{{ IDR }}/lstbin-outputs/{{ CASENAME }}/"
-bl_chunk_size = 5000
+bl_chunk_size = "${LSTBIN_OPTS:bl_chunk_size}"
```
Comment (jsdillon) [inline pipelines/h6c/idr2/v3/lstbin/lstbin-template.toml:63]: Why have this at all? I'd just pass in ${LSTBIN_OPTS:bl_chunk_size} where you'd previously pass in ${LSTAVG_OPTS:bl_chunk_size}

File: src/hera_pipelines/cli.py [hunk 3]
```diff
@@ -253,7 +245,8 @@ def lstbin_setup(season, idr, generation, repodir, cases, force, setup_analysis,
             CALEXT=".smooth.calfits" if abscal == "smoothcal" and redavg=='nonavg' else (
                 ".abs.calfits" if abscal=='abscal' and redavg=='nonavg' else 'none'
             ),
-            FLAGGED_AVERAGE = dlyfilt=='dlyfilt'
+            FLAGGED_AVERAGE = dlyfilt=='dlyfilt',
+            REDAVG=redavg == 'redavg',
```
Comment (jsdillon) [inline src/hera_pipelines/cli.py]: What's going on with the whitespace here?

File: pipelines/h6c/idr2/v3/lstbin/lstbin-template.toml (hunk 1 above)
Comment (steven-murray) [reply pipelines/h6c/idr2/v3/lstbin/lstbin-template.toml:34]: AH yeah, this is a Jinja2 template that gets filled out by a CLI command. This is explained in the comment at the top of the TOML. I've added a bit more information to it.

File: pipelines/h6c/idr2/v3/lstbin/lstbin-template.toml (hunk 2 above)
Comment (steven-murray) [reply pipelines/h6c/idr2/v3/lstbin/lstbin-template.toml:63]: The `bl_chunk_size` parameter needs to be used both by `hera_opm` and also the notebook itself as a parameter. These both use different parts of this TOML, so I put it into both.

File: src/hera_pipelines/cli.py (hunk 3 above)
Comment (steven-murray) [reply src/hera_pipelines/cli.py]: haha no idea....

**Review by jsdillon — APPROVED** [review]

**Pull request merged**